File: esbmc.py/Testes02/validador_titulo/validador_nao_ser_espacos_titulo.py
MAX_LEN_TITLE: int = 21

def validate_not_only_spaces(title: str, length: int) -> bool:
    only_spaces: bool = True
    
    i: int = 0
    while i < length:
        if title[i] != ' ':
            only_spaces = False
            break
        i += 1

    assert not only_spaces, "O título não pode ser apenas espaços."
    return True


def main() -> None:
    title: str = input()    
    length: int = len(title)

    assert 1 <= length <= 20  

    validate_not_only_spaces(title, length)


main()

File: esbmc.py/Testes02/validador_titulo/validador_nao_ser_espacos_titulo.cpp
#define MAX_LEN_TITLE 21

using namespace std;

bool validate_not_only_spaces(const char title[MAX_LEN_TITLE], int len) {
    bool only_spaces = true;
    for (int i = 0; i < len; i++) {
        if (!isspace(title[i])) {
            only_spaces = false;
            break;
        }
    }
    __ESBMC_assert(!only_spaces, "O título não pode ser apenas espaços.");
    return true;
}

int main() {
    char title[MAX_LEN_TITLE];
    int len;

    __ESBMC_assume(len >= 1 && len <= 20);

    validate_not_only_spaces(title, len);

    return 0;
}
